pkg/controllers/federatedcluster/controller.go

```diff
@@ -457,12 +458,16 @@ func (c *FederatedClusterController) handleTerminatingCluster(
 	}
 
 	// We have already checked that we are the last finalizer so we can simply set finalizers to be empty.
-	cluster.SetFinalizers(nil)
-	if _, err := c.fedClient.CoreV1alpha1().FederatedClusters().Update(
-		ctx,
-		cluster,
-		metav1.UpdateOptions{},
-	); err != nil {
+	if err := retry.RetryOnConflict(retry.DefaultBackoff, func() error {
+		var err error
+		cluster, err = c.fedClient.CoreV1alpha1().FederatedClusters().Get(ctx, cluster.Name, metav1.GetOptions{ResourceVersion: "0"})
```

Comment:
Should we only try to get cluster again when we actually got a 409?

Reply:
PTAL.